### PR TITLE
Added methods for getting untracked and uncommitted python files

### PR DESCRIPTION
**Pull request by TamerSherif**
### Why is this PR necessary?
This PR is intended to allow us to fetch any python files that are uncommitted or untracked. This was brought up while working on https://github.com/Shopify/starscream/pull/37131. While attempting to move the linting to be achieved through a `.py` module we have the ability to use `shopify_python` in order to fetch the changed files to run the linter on. Currently `shopify_python` does not support getting untracked and uncommitted files.

### How does this PR work?
This PR adds two additional methods allowing us to have three methods in total that provide us will all changed python files in a repo.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -2,5 +2,7 @@
 *.egg-info
 .DS_Store
 .cache
+# ignore pycharm config files
```
Comment (alejandrolujan) [inline .gitignore]: Good addition, let's add this on a separate PR to keep things clean.

File: pylintrc [hunk 2]
```diff
@@ -38,7 +38,7 @@ load-plugins=shopify_python
 # redefined-outer-name: doesn't play well with pytest fixtures
 # bad-continuation: PyLint 2.0 fixes expected indentation of wrapped `with` lines. Disable this rule to avoid triggering
 #     false errors in PyLint 1.9.
-disable=missing-docstring,redefined-outer-name,bad-continuation
+disable=missing-docstring,redefined-outer-name,bad-continuation,redefined-builtin
```
Comment (alejandrolujan) [inline pylintrc]: Why was this added?

File: shopify_python/git_utils.py [hunk 3]
```diff
@@ -60,6 +59,20 @@ def changed_python_files_in_tree(root_path):
...
 
+def uncommitted_python_files(root_path):
+    # type: (str) -> typing.FrozenSet[str]
+    git_repo = repo.Repo(root_path)
+    unstaged_files = [file.a_path for file in git_repo.index.diff(None) if _file_is_python(file.a_path)]
+    staged_files = [file.a_path for file in git_repo.index.diff('HEAD') if _file_is_python(file.a_path)]
+    return frozenset(unstaged_files + staged_files)
+
+
+def untracked_python_files(root_path):
```
Comment (alejandrolujan) [inline shopify_python/git_utils.py]: We need a test for this method as well.

File: shopify_python/git_utils.py [hunk 4]
```diff
@@ -60,6 +59,20 @@ def changed_python_files_in_tree(root_path):
             if os.path.exists(abs_mod) and os.path.isfile(abs_mod) and _file_is_python(abs_mod)]
 
 
+def uncommitted_python_files(root_path):
```
Comment (alejandrolujan) [inline shopify_python/git_utils.py]: Let's add a docstring to both of these methods to explain what they do.

Comment (alejandrolujan) [reply shopify_python/git_utils.py]: Also, why group unstaged and staged files in one method, and keep untracked separate?

Perhaps it would be better to split these into two, and then add one more method to union them, something like this:

- `untracked_python_files `
- `untstaged_python_files `
- `staged_python_files `
- `changed_local_python_files` (union the three above)

File: tests/shopify_python/test_git_utils.py [hunk 5]
```diff
@@ -211,6 +211,15 @@ def test_dont_include_uncommited_or_untracked_files(main_repo, python_file):
     assert git_utils.changed_python_files_in_tree(main_repo.working_dir) == [os.path.basename(python_file)]
 
 
+def test_get_uncommitted_python_files(main_repo, python_file):
```
Comment (alejandrolujan) [inline tests/shopify_python/test_git_utils.py]: Here you're only testing with staging a file, we should also test unstaged files.

File: .gitignore (hunk 1 above)
Comment (TamerSherif) [reply .gitignore]: Here it is: https://github.com/Shopify/shopify_python/pull/111

File: pylintrc (hunk 2 above)
Comment (TamerSherif) [reply pylintrc]: It's for builtins being used as variable names. I removed it now. I just changed my loop so it avoids using the variable name `file` which is a Python builtin.

File: shopify_python/git_utils.py [hunk 6]
```diff
@@ -51,7 +51,7 @@ def _file_is_python(path):
 
 def changed_python_files_in_tree(root_path):
     # type: (str) -> typing.List[str]
-
+    """Gets a list of paths of all committed files in a given repository."""
```
Comment (alejandrolujan) [inline shopify_python/git_utils.py]:
```suggestion
    """Returns the list of all committed files in a given repository when compared with the remote master."""
```

File: shopify_python/git_utils.py [hunk 7]
```diff
@@ -60,6 +60,41 @@ def changed_python_files_in_tree(root_path):
             if os.path.exists(abs_mod) and os.path.isfile(abs_mod) and _file_is_python(abs_mod)]
 
 
+def unstaged_python_files(root_path):
+    # type: (str) -> typing.List[str]
+    """Gets a list of paths of all uncommitted unstaged files in a given repository."""
```
Comment (alejandrolujan) [inline shopify_python/git_utils.py]:
```suggestion
    """Returns the list of all unstaged files in a given repository."""
```

File: shopify_python/git_utils.py [hunk 8]
```diff
@@ -60,6 +60,41 @@ def changed_python_files_in_tree(root_path):
...
+    git_repo = repo.Repo(root_path)
+    return [changed_file.a_path for changed_file in git_repo.index.diff(None) if _file_is_python(changed_file.a_path)]
+
+
+def staged_python_files(root_path):
+    # type: (str) -> typing.List[str]
+    """Gets a list of paths of all uncommitted staged files in a given repository."""
```
Comment (alejandrolujan) [inline shopify_python/git_utils.py]:
```suggestion
    """Returns the list of all staged files in a given repository."""
```

File: shopify_python/git_utils.py [hunk 9]
```diff
@@ -60,6 +60,41 @@ def changed_python_files_in_tree(root_path):
...
+    git_repo = repo.Repo(root_path)
+    return [changed_file.a_path for changed_file in git_repo.index.diff('HEAD') if _file_is_python(changed_file.a_path)]
+
+
+def untracked_python_files(root_path):
+    # type: (str) -> typing.List[str]
+    """Gets a list of paths of all untracked files in a given repository."""
```
Comment (alejandrolujan) [inline shopify_python/git_utils.py]:
```suggestion
    """Returns the list of all untracked files in a given repository."""
```

File: shopify_python/git_utils.py [hunk 10]
```diff
@@ -60,6 +60,41 @@ def changed_python_files_in_tree(root_path):
...
+    git_repo = repo.Repo(root_path)
+    return [item for item in git_repo.untracked_files if _file_is_python(item)]
+
+
+def changed_local_python_files(root_path):
+    # type: (str) -> typing.FrozenSet[str]
+    """
+    Gets a list of paths of all changed files.
```
Comment (alejandrolujan) [inline shopify_python/git_utils.py]:
```suggestion
    Returns the list of all changed files.
```
